### ISSUE TITLE
asset dir support in commands

### ISSUE THREAD

**Issue by bpoldrack** [psyinfra/onyo#515]
Asset dirs (a path that is both - an asset and a directory) are implemented in principle and supported by the inventory operations. An asset dir has two dot files: the `.anchor` because it's just like any other inventory dir, and an `.onyo-asset-dir` YAML file. The name of that directory obviously is generated form that YAML, because it's an asset.

Consequently the operations treat that directory as a thing with two aspects: It serves as a valid asset identifier as well as a regular inventory directory. That implies that all the operations on directories and all the operations for assets need to be able to deal with such a thing. `rename_directory` should fail, because it's an asset (and the name needs to be generated). Everything ales can work in principle. The interesting part is adding/removing an asset or directory pointing at that path.
What the operations do is: Add/Remove only that aspect of that path. So, `add_asset` being pointed to a regular directory, turns it into an asset dir. As does `add_directory` being pointed to an asset file. Same thing the other way around with removal.

This makes sure, that the operations record reflects the concept without making asset dirs a special, third type of thing an inventory tracks. Instead, it's two things that happen to use the same identifier/label (path), but one could replay the operations in a DB, where assets and directories are stored differently and independently. One would get the same thing: A directory and an asset using the same "name".

However, none of that defines what actual onyo commands should do yet.
Long story short: Define and implement command behavior WRT asset dirs.
